src/main/proto/netflix/titus/agent.proto

```diff
@@ -269,4 +269,54 @@ message ContainerInfo {
     }
     // Process to be executed inside the container
     optional Process process = 38;
+
+    // Container hostname (not necessarily the same as task ID)
+    optional string hostName = 39;
+
+    // Container task ID
+    optional string taskId = 40;
+
+    // Container launch timestamp, in Unix epoch seconds
+    optional uint64 launchTimeUnixSec = 41;
+}
+
+// For the identity client to query the metadata server to confirm the
+// currently running task's identity
+message TaskIdentity {
+    // Current state of the task
+    required messages.TaskInfo task = 1;
+
+    // Container details, including:
+    //  - app name, stack, detail
+    //  - image name, digest, label
+    //  - environment variables
+    //  - command, entrypoint
+    //  - metadata string and signature from caller
+    required messages.ContainerInfo container = 2;
+
+    // Timestamp, in Unix epoch seconds, when request was made.
+    required uint64 unixTimestampSec = 3;
+
+    // IP address of container
+    required string ipv4Address = 4;
+}
+
+message CertificateSignature {
+    // Raw signature produced from signing input bytes using the
+    // private key of the titus agent’s certificate.
+    required bytes signature = 1;
+
+    // Name of signature algorithm used (e.g.” SHA512withRSAandMGF1”)
```

Comment:
Can this be an enum?

Reply:
@JackOfMostTrades and @douglasc-nflx can weigh in here - I don't think it's an enum in their services, so I'm matching that behaviour.

Reply:
Yea, we treat it as a string, but there's no reason it couldn't change it to be an enum. I think we actually end up converting the string back to an enum on our end anyway. At present the supported values are the following if you want to put these into the protobuf enum:
```
SHA256withRSAandMGF1
SHA384withRSAandMGF1
SHA512withRSAandMGF1
SHA256withECDSA
SHA384withECDSA
SHA512withECDSA
```

Reply:
@JackOfMostTrades - enum'ed.  Let me know if that works for you.

Reply:
Looks good to me.